pokedex/model/handwritten/states.asl

```diff
@@ -140,11 +159,46 @@ begin
   MSTATUS_MPP = __BitsToPrivLevel(value);
 end
 
+getter MSTATUS_FS => bits(2)
+begin
+  return _MSTATUS_FS;
+end
+
+setter MSTATUS_FS = value : bits(2)
+begin
+  _MSTATUS_FS = value;
+
+  _internal_set_mstatus_sd();
+end
+
+getter MSTATUS_VS => bits(2)
+begin
+  return _MSTATUS_VS;
+end
+
+setter MSTATUS_VS = value : bits(2)
+begin
+  _MSTATUS_VS = value;
+
+  _internal_set_mstatus_sd();
+end
+
+func _internal_set_mstatus_sd()
+begin
+  // TODO: add VS, XS when supported
+  if _MSTATUS_FS == '11' || _MSTATUS_VS == '11' then
+    MSTATUS_SD = '1';
+  elsif _MSTATUS_FS != '11' && _MSTATUS_VS != '11' then
+    MSTATUS_SD = '0';
+  end
+end
```

Comment:
I prefer to compute mstatus.SD on the fly at reading, instead of syncing it in every write.